include/reactor-cpp/reactor.hh

```diff
@@ -52,6 +52,7 @@ public:
   void shutdown() final;
 
   virtual void assemble() = 0;
+  virtual void construct() {}
```

Comment:
Okay, here is some food for thought:

Currently, the construction phase if for instantiating all the reactors. Here you introduce this new `construct` method that reactors need to implement, so the question: what should go into the constructor of the class and what should go into this `construct` function.

and `sed /{}/=default`